### ISSUE TITLE
ipfs get does not support zip compression

### ISSUE THREAD

**Issue by chenminjian** [ipfs/go-ipfs#6104]
#### Version information:
go-ipfs version: 0.4.19-dev-
Repo version: 7
System version: amd64/darwin
Golang version: go1.11.1

#### Type:
enhancement

#### Description:

`ipfs get` does not support zip compression. Zip is used more widely in windows.

**Comment by magik6k** [comment]
This should be relatively simple to do:
1. Implement `ZipWriter` in go-ipfs-files based on https://github.com/ipfs/go-ipfs-files/blob/master/tarwriter.go#L16
2. Add `--zip` option to `ipfs got` and make it swap [this](https://github.com/ipfs/go-ipfs/blob/master/core/commands/get.go#L319) to the zip writer

**Comment by Kubuxu** [comment]
Do we really need this feature? One of the main reasons why we have a tar writer is because we use tar format for API we essentially get tar archive output for free by writing the API output to a file (and maybe compressing it on the way).

**Comment by chenminjian** [comment]
Yes, we use tar for writing the API output to a file. But in some windows systems, we can't open tar.

**Comment by lidel** [comment]
FWIW support for tar has been included in Microsoft Windows since Windows 10 (2019):
https://techcommunity.microsoft.com/t5/containers/tar-and-curl-come-to-windows/ba-p/382409
